dm/dm/worker/source_worker_test.go

```diff
@@ -859,12 +878,24 @@ func (t *testServer) TestGetWorkerValidatorErr(c *C) {
 	w := t.setupValidator(c)
 	// when subtask name not exists
 	// return empty array
-	c.Assert(len(w.GetWorkerValidatorErr("invalidTask", pb.ValidateErrorState_InvalidErr)), Equals, 0)
+	errors, err := w.GetWorkerValidatorErr("invalidTask", pb.ValidateErrorState_InvalidErr)
+	c.Assert(err.Error(), Matches, ".*sub task with name.*not found.*")
+	c.Assert(errors, IsNil)
 	// subtask match
-	c.Assert(len(w.GetWorkerValidatorErr("testQueryValidator", pb.ValidateErrorState_InvalidErr)), Equals, 2)
+	errors, err = w.GetWorkerValidatorErr("testQueryValidator", pb.ValidateErrorState_InvalidErr)
+	c.Assert(err, IsNil)
+	c.Assert(len(errors), Equals, 2)
 }
 
 func (t *testServer) TestOperateWorkerValidatorErr(c *C) {
+	c.Assert(failpoint.Enable("github.com/pingcap/tiflow/dm/syncer/MockValidationQuery", `return(true)`), IsNil)
+	c.Assert(failpoint.Enable("github.com/pingcap/tiflow/dm/syncer/MockValidationOperation", `return(true)`), IsNil)
+	c.Assert(failpoint.Enable("github.com/pingcap/tiflow/dm/dm/worker/MockValidationQuery", `return(true)`), IsNil)
```

Comment:
i don't like the idea of using 3 failpoints only to test a 4-5 lines method(with only `if` branch), let integration test cover the normal case, only need to test the error case using unit test.

TestGetWorkerValidatorErr too